### PR TITLE
Update Java SDK Documentation

### PR DESCRIPTION
**Pull request by aarsilv**
Update the Java SDK Documentation to account for its migration to UFC and for bringing its Bandit methodology up to speed with all the changes that happened since this first implementation.

### CODE REVIEW

File: docs/sdks/server-sdks/java.md [hunk 1]
```diff
@@ -12,270 +12,374 @@ Eppo's open source Java SDK can be used for both feature flagging and experiment
...
-        .properties(ImmutableMap.builder()
-            .put("experiment", event.experiment)
-            .put("variation", event.variation)
-            .put("timestamp", event.timestamp)
-            .build()
-        )
-    );
-  }
-}
+String assignedVariation = eppoClient.getStringAssignment("subjectKey", "flagkey", "defaultValue");
```
Comment (schmit) [inline docs/sdks/server-sdks/java.md:67]: Can a non-optional argument follow an optional attribute? How does the compiler know "defaultValue" is not subjectAttributes?

File: docs/sdks/server-sdks/java.md [hunk 2]
```diff
@@ -12,270 +12,374 @@ Eppo's open source Java SDK can be used for both feature flagging and experiment
...
+The properties of the event object passed to the assignment logger, accessible via getters, are as follows:
 
-- `subjectKey` - The entity ID that is being experimented on, typically represented by a uuid.
-- `flagOrExperimentKey` - This key is available on the detail page for both flags and experiments.
-- `targetingAttributes` - An optional map of metadata about the subject used for targeting. If you create rules based on attributes on a flag/experiment, those attributes should be passed in on every assignment call.
+| Field                                | Description                                                                                                                | Example                                |
+|--------------------------------------|----------------------------------------------------------------------------------------------------------------------------|----------------------------------------|
+| `timestamp` (Date)                   | The time when the subject was assigned to the variation                                                                    | Mon Aug 19 21:46:02 UTC 2024           |
```
Comment (schmit) [inline docs/sdks/server-sdks/java.md:129]: I imagine the timestamp is not actually `Mon Aug 19 21:46:02 UTC 2024` 😅  -- should we put the actual string value?

File: docs/sdks/server-sdks/java.md [hunk 3]
```diff
@@ -12,270 +12,374 @@ Eppo's open source Java SDK can be used for both feature flagging and experiment
...
+| `subject` (String)                   | The identifier of the subject (e..g, user) assigned to the experiment variation                                            | "695e8121-96dc-4185-aedd-ef40225a2ef2" |
+| `subjectAttributes` (Attributes)     | A free-form map of metadata about the subject.                                                                             | {country=FR, age=60}                   |
+| `extraLogging` (Map<String, String>) | Any extra information relevant to the assignment                                                                           | {holdout=q1-holdout}                   |
+| `metaData` (Map<String, String>)     | Any additional freeform meta data, such as the version of the SDK                                                          | {sdkLibVersion=3.0.1}                  |
 
-![Example flag key](/img/feature-flagging/flag-key.png)
+Note that the `Attributes` type is an extension of `Map<String, EppoValue>`, with `EppoValue` being a container for values 
+that have strings, numbers, booleans, or JSON.
+
+:::note
```
Comment (schmit) [inline docs/sdks/server-sdks/java.md:142]:
```suggestion
:::info
```

makes it stands out more

**Review by schmit — APPROVED** [review]
Nice!

File: docs/sdks/server-sdks/java.md (hunk 1 above)
Comment (aarsilv) [reply docs/sdks/server-sdks/java.md:67]: That's the beauty of strongly-typed languages like Java. It looks at method _signatures_. So if it sees three `String`s, it knows no attributes are provided. If it sees `String, String, Attributes, String`, it knows attributes are provided. `String, String, Attributes` would not be allowed--default must always be provided.

All that being said, I like loosely typed languages like JavaScript. But this specific case of overloading methods is a perk.

File: docs/sdks/server-sdks/java.md (hunk 2 above)
Comment (aarsilv) [reply docs/sdks/server-sdks/java.md:129]: The JAVA SDK passes `Date` objects around (which hold date and time), so for the example value, I put the `toString()` value of a Java date object.

File: docs/sdks/server-sdks/java.md [hunk 4]
```diff
@@ -12,270 +12,374 @@ Eppo's open source Java SDK can be used for both feature flagging and experiment
...
+| `subject` (String)                   | The identifier of the subject (e..g, user) assigned to the experiment variation                                            | "695e8121-96dc-4185-aedd-ef40225a2ef2" |
+| `subjectAttributes` (Attributes)     | A free-form map of metadata about the subject.                                                                             | {country=FR, age=60}                   |
+| `extraLogging` (Map<String, String>) | Any extra information relevant to the assignment                                                                           | {holdout=q1-holdout}                   |
+| `metaData` (Map<String, String>)     | Any additional freeform meta data, such as the version of the SDK                                                          | {sdkLibVersion=3.0.1}                  |
 
-![Example flag key](/img/feature-flagging/flag-key.png)
+Note that the `Attributes` type is an extension of `Map<String, EppoValue>`, with `EppoValue` being a container for values
```
[the remark this answers is not in the document]
Comment (aarsilv) [reply docs/sdks/server-sdks/java.md:139]: oh good call!

File: docs/sdks/server-sdks/java.md (hunk 3 above)
Comment (aarsilv) [reply docs/sdks/server-sdks/java.md:142]: didn't know about that! thanks!

**Pull request merged**